frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/task-sidebar/TaskSidebarResource.tsx

```diff
@@ -12,28 +12,49 @@ type TaskSidebarResourceProps = {
   resource: TektonResource;
 };
 
+interface ResourceLink {
+  name: string;
+  resource: string;
+}
+
 const TaskSidebarResource: React.FC<TaskSidebarResourceProps> = (props) => {
   const { t } = useTranslation();
-  const { setFieldValue } = useFormikContext<PipelineBuilderFormikValues>();
+  const { getFieldMeta, setFieldValue } = useFormikContext<PipelineBuilderFormikValues>();
   const {
     availableResources,
     hasResource,
     name,
-    resource: { name: resourceName, type: resourceType },
+    resource: { name: resourceName, type: resourceType, optional = false },
   } = props;
 
   const dropdownResources = availableResources.filter(
     (resource) => resourceType === resource.type && !!resource.name?.trim(),
   );
+
+  const currentLinkedResourceName = getFieldMeta<ResourceLink>(name).value?.resource;
+  const currentLinkedResourceSelectable =
+    currentLinkedResourceName &&
+    dropdownResources.some((resource) => resource.name === currentLinkedResourceName);
+
   const options: FormSelectFieldOption[] = [
     {
       label: t('pipelines-plugin~Select {{resourceType}} resource...', { resourceType }),
       value: '',
       isPlaceholder: true,
-      isDisabled: true,
     },
-    ...dropdownResources.map((resource) => ({ label: resource.name, value: resource.name })),
   ];
+  if (currentLinkedResourceName && !currentLinkedResourceSelectable) {
+    options.push({
+      label: t('pipelines-plugin~{{resourceName}} (deleted)', {
+        resourceName: currentLinkedResourceName,
+      }),
+      value: currentLinkedResourceName,
+      isDisabled: true,
+    });
```

Comment:
This may be a QoL but this is a change in the way we manage the dropdown, thus a feature. I think we need to roll back to `Select...` like we do with everything else. The error under the field informs what happened.